src/compile.rs

```diff
@@ -2084,49 +2118,74 @@ pub fn call_derived_associated_calls<'a>(
-                                f.declaration_file,
-                            ) {
-                                Ok(c) => c.inner(&mut warnings),
-                                Err(e) => {
-                                    errors.append(e);
-                                    continue;
-                                }
-                            };
 
-                            // These are definitely ValidatedCalls, but ValidatedCall::new()
-                            // returns ValidatedStatements, so they can be inserted directly
-                            for c in validated_calls {
-                                ret.insert(c);
+        for f in funcs.values() {
+            if f.is_associated_call {
+                let resource_name = match f.class {
+                    FunctionClass::Type(n) => n.name.clone(),
+                    _ => {
+                        // Can't derive from Global or API
+                        continue;
+                    }
+                };
+                // Calls from annotations were already added if they weren't derived.  If the call
+                // was derived AND via an annotation, we add it.  If it was via nest, we add it
+                // unconditionally
+                let mut nested_associate = false;
+                let mut regular_associate = false;
+                for ann in &t.annotations {
+                    match ann {
+                        AnnotationInfo::Associate(associations) => {
+                            if associations
+                                .resources
+                                .iter()
+                                .any(|r| get_synthetic_resource_name(&t.name, r) == resource_name)
+                            {
+                                regular_associate = true;
                             }
                         }
+                        AnnotationInfo::NestAssociate(associations) => {
+                            if associations.resources.iter().any(|r| r == &resource_name) {
+                                nested_associate = true;
+                            }
+                        }
+                        _ => (),
+                    }
+                }
+                if regular_associate || nested_associate {
+                    let r_basename = match resource_name.as_ref().split_once('.') {
+                        Some((_, r)) => r,
+                        None => resource_name.as_ref(),
+                    };
+                    let call = make_associated_call(
+                        get_synthetic_resource_name(&t.name, &r_basename.into()),
+                        f,
+                    );
+                    let args = vec![FunctionArgument::new_this_argument(t)];
+                    // TODO: Should there be a parent_context here?  I think this is
+                    // effectively a "fake" context since we're not really parsing the tree
```

Comment:
What are you thinking the parent context should be?

Reply:
So the only real change here is indent.  The comment existed both with and without this change.

I'm not really sure that having a real context actually makes sense though.  Having a context for the domain would let us handle let bindings from the domain.  But we may not want that anyways.  I'm not sure that something other than this fake context is really needed?  Idk...